internal/controller/package_controller.go

```diff
@@ -230,13 +230,17 @@ func (r *PackageReconcilationContext) ensureDependencies(ctx context.Context) bo
 	log.V(1).Info("ensuring dependencies", "dependencies", r.pi.Status.Manifest.Dependencies)
 
 	var failed []string
-	if result, err := r.dependencyMgr.Validate(ctx, r.pi.Status.Manifest); err != nil {
+	if result, err := r.dependencyMgr.Validate(ctx, r.pi.Status.Manifest, r.pkg.Spec.PackageInfo.Version); err != nil {
 		r.setShouldUpdate(
 			conditions.SetFailed(ctx, r.EventRecorder, r.pkg, &r.pkg.Status.Conditions,
 				condition.InstallationFailed, fmt.Sprintf("error validating dependencies: %v", err)))
 		return false
 	} else if result.Status == dependency.ValidationResultStatusResolvable {
 		for _, requirement := range result.Requirements {
+			if requirement.Transitive {
+				// Only direct dependencies should be touched in the context of the reconciliation of a package.
+				continue
```

Comment:
Hm this is probably a bug. In your example, if cert-manager is a dependency of keptn, transitive should be false and it should be installed. If cert-manager had itself a dependency, then that requirement should be marked as transitive and not installed during the reconciliation of the keptn package (because it is not a dependency of keptn).

Reply:
Should be fixed now, thanks for finding this!